File: sayit/sayit.js
// My first foray into Node.js.
// Basically a direct port of my python offering.
// I'd like to find a way to avoid using a global variable.
/* Also, I thought writing a javascript solution would show me how to
   implement a C solution, but I'm just as clueless as ever. */

// Global for storing invoked parameters
var words = [];

function sayIt(word) {
    // Check to see if a parameter was given.
    if (word == null) {
        // If no parameter, empty words and print.
        message = words.join(' ');
        words = [];
        console.log(message);
        return message;
    } else {
        // If a parameter is given, push to words and repeat.
        words.push(word);
        return sayIt;
    }
}

sayIt("Hey")("there!")()

File: sayit/sayit.c
// This is garbage that won't compile.
// I don't even know if functions an be return values (line 19).
# include <stdio.h>

// Global for storing invoked parameters
char words[];

sayIt( char word[] ) {
    // Check to see if a parameter was given.
    if (word[0] == '\0') {
        // If no parameter, empty words and print.
        char message = words;
        words[0] = '\0';
        printf(message);
        return message;
    } else {
        // If a parameter is given, push to words and repeat.
        words = words + ' ' + word
        return sayIt;
    }
}

sayIt("Hey")("there!")()
